src/test/lib/async-append_test.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+/// <reference path="../../../node_modules/@types/mocha/index.d.ts" />
+/// <reference path="../../../node_modules/@types/chai/index.d.ts" />
+
+import {asyncAppend} from '../../lib/async-append.js';
+import {html, render} from '../../lit-html.js';
+
+import {TestAsyncIterable} from './test-async-iterable.js';
+
+const assert = chai.assert;
+
+// Set Symbol.asyncIterator on browsers without it
+if (typeof Symbol !== undefined && Symbol.asyncIterator === undefined) {
+  Object.defineProperty(Symbol, 'asyncIterator', {value: Symbol()});
+}
+
+suite('asyncAppend', () => {
```

Comment:
I would like to see a test here for rendering lists of element nodes. It seems like a common case, and also one that has tripped us up in the past (particularly when it comes to getting the intersections between two lists right).